services/cd-service/pkg/service/tags.go

```diff
@@ -68,15 +68,12 @@ func (s *TagsServer) GetProductSummary(ctx context.Context, in *api.GetProductSu
 	for _, row := range summaryFromEnv {
 		for _, app := range response.Applications {
 			if row.App == app.Name {
-				length := len(app.Releases) - 1
-				if app.Releases[length].DisplayVersion != "" {
-					productVersion = append(productVersion, &api.ProductSummary{App: row.App, Version: app.Releases[length].DisplayVersion})
-				} else if app.Releases[length].SourceCommitId != "" {
-					productVersion = append(productVersion, &api.ProductSummary{App: row.App, Version: app.Releases[length].SourceCommitId})
-				} else {
-					productVersion = append(productVersion, &api.ProductSummary{App: row.App, Version: row.Version})
+				for _, release := range app.Releases {
+					if strconv.Itoa(int(release.Version)) == row.Version {
+						productVersion = append(productVersion, &api.ProductSummary{App: row.App, Version: row.Version, CommitId: release.SourceCommitId, DisplayVersion: release.DisplayVersion})
+						break
+					}
 				}
-				break
 			}
 		}
 	}
```

Comment:
please sort the productionVersions, e.g. by app name, e.g. `sort.Strings(...)` or `sort.Sort(...)`.
As of now the order is undefined.